cloud/amazon/ec2_vol.py

```diff
@@ -325,8 +392,40 @@ def attach_volume(module, ec2, volume, instance):
         except boto.exception.BotoServerError, e:
             module.fail_json(msg = "%s: %s" % (e.error_code, e.error_message))
 
+        modify_dot_attribute(module, ec2, instance, device_name)
+
     return volume, changed
 
+
+def modify_dot_attribute(module, ec2, instance, device_name):
+
+    delete_on_termination = module.params.get('delete_on_termination')
+    changed = False
+
+    try:
+        instance.update()
+        dot = instance.block_device_mapping[device_name].delete_on_termination
+    except boto.exception.BotoServerError, e:
+        module.fail_json(msg = "%s: %s" % (e.error_code, e.error_message))
+  
+    if delete_on_termination ^ dot:
+        try:
+            bdt = BlockDeviceType(delete_on_termination=delete_on_termination)
+            bdm = BlockDeviceMapping()
+            bdm[device_name] = bdt
+
+            ec2.modify_instance_attribute(instance_id=instance.id, attribute='blockDeviceMapping', value=bdm)
+
+            while not (instance.block_device_mapping[device_name].delete_on_termination ^ delete_on_termination):
```

Comment:
Again, this is unclear. I think this is just
`while instance.block_device_mapping[device_name].delete_on_termination != delete_on_termination:`